src/Reader/XLSX/SheetHeaderReader.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenSpout\Reader\XLSX;
+
+use OpenSpout\Common\Exception\IOException;
+use OpenSpout\Reader\Common\ColumnWidth;
+use OpenSpout\Reader\Common\XMLProcessor;
+use OpenSpout\Reader\Wrapper\XMLReader;
+
+final class SheetHeaderReader
+{
+    public const XML_NODE_COL = 'col';
+    public const XML_NODE_SHEETDATA = 'sheetData';
+    public const XML_ATTRIBUTE_MIN = 'min';
+    public const XML_ATTRIBUTE_MAX = 'max';
+    public const XML_ATTRIBUTE_WIDTH = 'width';
+
+    /** @var string Path of the XLSX file being read */
+    private string $filePath;
+
+    /** @var string Path of the sheet data XML file as in [Content_Types].xml */
+    private string $sheetDataXMLFilePath;
+
+    /** @var XMLReader The XMLReader object that will help read sheet's XML data */
+    private XMLReader $xmlReader;
+
+    /** @var XMLProcessor Helper Object to process XML nodes */
+    private XMLProcessor $xmlProcessor;
+
+    /** @var ColumnWidth[] The widths of the columns in the sheet, if specified */
+    private array $columnWidths = [];
+
+    /**
+     * @param string       $filePath             Path of the XLSX file being read
+     * @param string       $sheetDataXMLFilePath Path of the sheet data XML file as in [Content_Types].xml
+     * @param XMLReader    $xmlReader            XML Reader
+     * @param XMLProcessor $xmlProcessor         Helper to process XML files
+     */
+    public function __construct(
+        string $filePath,
+        string $sheetDataXMLFilePath,
+        XMLReader $xmlReader,
+        XMLProcessor $xmlProcessor
+    ) {
+        $this->filePath = $filePath;
+        $this->sheetDataXMLFilePath = $this->normalizeSheetDataXMLFilePath($sheetDataXMLFilePath);
+        $this->xmlReader = $xmlReader;
+
+        // Register all callbacks to process different nodes when reading the XML file
+        $this->xmlProcessor = $xmlProcessor;
+        $this->xmlProcessor->registerCallback(self::XML_NODE_COL, XMLProcessor::NODE_TYPE_START, [$this, 'processColStartingNode']);
+        $this->xmlProcessor->registerCallback(self::XML_NODE_SHEETDATA, XMLProcessor::NODE_TYPE_START, [$this, 'processSheetDataStartingNode']);
```

Comment:
Infection says that removing this line doesn't result in any test failing.
I'm fine with the other 3 Infection reports not being addressed, but this one puzzles me.

If no test fails, what's the purpose of this callback?

Reply:
The reason I added it is because of performance. Once you get to the `<sheetData>` element, the header reading should stop: the row data starts. If you have a very large sheet with a lot of rows, it makes no sense to read all those rows when there is no header information left.

So on lines 63/64, I do:
```
        // Now read the entire header of the sheet, until we reach the <sheetData> element
        $this->xmlProcessor->readUntilStopped();
```

By setting this callback, the header processing stops at the `<sheetData>` element, and you can close the reader. 

I don't know how to write a test for that though.